File: EMS Advanced Data Import 3.13 Full Source for Rad Studio 11 Alexandria/demos/Delphi/CustomDestination/Unit1.pas
unit Unit1;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  Qimport3, Qimport3XLS, Grids, StdCtrls;

type
  TForm1 = class(TForm)
    Button1: TButton;
    StringGrid1: TStringGrid;
    QImportXLS1: TQimport3XLS;
    Button2: TButton;
    procedure QImportXLS1UserDefinedImport(Sender: TObject;
      Row: TQImportRow);
    procedure QImportXLS1BeforeImport(Sender: TObject);
    procedure FormCreate(Sender: TObject);
    procedure Button1Click(Sender: TObject);
    procedure Button2Click(Sender: TObject);
    procedure QImportXLS1AfterImport(Sender: TObject);
  private
    FCounter: integer;
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

{$R *.DFM}

var
  arCountry: array[0..4, 0..17] of string;

procedure TForm1.FormCreate(Sender: TObject);
var
  i: Integer;
  dbDir: string;
begin
  i := 0;
  dbDir := ExtractFileDir(Application.ExeName);
  while (not FileExists(dbDir + '\data\country.xls')) and (i < 4) do
  begin
    dbDir := ExtractFileDir(dbDir);
    Inc(i);
  end;
  QImportXLS1.FileName := dbDir + '\data\country.xls';
  StringGrid1.Cells[0, 0] := 'Name';
  StringGrid1.Cells[1, 0] := 'Capital';
  StringGrid1.Cells[2, 0] := 'Continent';
  StringGrid1.Cells[3, 0] := 'Area';
  StringGrid1.Cells[4, 0] := 'Population';

  StringGrid1.ColWidths[0] := 125;
  StringGrid1.ColWidths[1] := 70;
  StringGrid1.ColWidths[2] := 80;
end;

procedure TForm1.QImportXLS1BeforeImport(Sender: TObject);
var
  i, j: integer;
begin
  for i := 0 to 4 do
    for j := 0 to 17 do
      arCountry[i, j] := EmptyStr;
  FCounter := 0;
end;

procedure TForm1.QImportXLS1UserDefinedImport(Sender: TObject;
  Row: TQImportRow);
var
  i: integer;
begin
  for i := 0 to Row.Count - 1 do
    arCountry[i, FCounter] := Row[i].Value;
  Inc(FCounter);
end;

procedure TForm1.Button1Click(Sender: TObject);
begin
  QImportXLS1.Execute;
end;

procedure TForm1.Button2Click(Sender: TObject);
var
  i, j: integer;
begin
  for i := 0 to 4 do
    for j := 1 to 18 do
      StringGrid1.Cells[i, j] := EmptyStr;
end;

procedure TForm1.QImportXLS1AfterImport(Sender: TObject);
var
  i, j: integer;
begin
  for i := 0 to 4 do
    for j := 0 to 17 do 
      StringGrid1.Cells[i, j + 1] := arCountry[i, j];
end;

end.

File: EMS Advanced Data Import 3.13 Full Source for Rad Studio 11 Alexandria/demos/CBuilder/CustomDestination/Unit1.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Unit1.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "QImport3"
#pragma link "QImport3XLS"
#pragma resource "*.dfm"
TForm1 *Form1;
String arCountry[5][18];
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TForm1::FormCreate(TObject *Sender)
{
  int i = 0;
  String dbDir = ExtractFileDir(Application->ExeName);
  while (!(FileExists(dbDir + "\\data\\country.xls")) && (i < 4))
  {
    dbDir = ExtractFileDir(dbDir);
    i++;
  }
  QImportXLS1->FileName = dbDir + "\\data\\country.xls";
  StringGrid1->Cells[0][0] = "Name";
  StringGrid1->Cells[1][0] = "Capital";
  StringGrid1->Cells[2][0] = "Continent";
  StringGrid1->Cells[3][0] = "Area";
  StringGrid1->Cells[4][0] = "Population";

  StringGrid1->ColWidths[0] = 125;
  StringGrid1->ColWidths[1] = 70;
  StringGrid1->ColWidths[2] = 80;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::QImportXLS1BeforeImport(TObject *Sender)
{
  for (int i = 0; i <= 4; i++)
    for (int j = 0; j <= 17; j++)
      arCountry[i][j] = EmptyStr;
  FCounter = 0;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::QImportXLS1UserDefinedImport(TObject *Sender,
  TQImportRow *Row)
{
  for (int i = 0; i <= Row->Count - 1; i++)
    arCountry[i][FCounter] = Row->Items[i]->Value;
  FCounter++;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button1Click(TObject *Sender)
{
  QImportXLS1->Execute();
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button2Click(TObject *Sender)
{
  for (int i = 0; i <= 4; i++)
    for (int j = 1; j <= 18; j++)
      StringGrid1->Cells[i][j] = EmptyStr;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::QImportXLS1AfterImport(TObject *Sender)
{
  for (int i = 0; i <= 4; i++)
    for (int j = 0; j <= 17; j++)
      StringGrid1->Cells[i][j + 1] = arCountry[i][j];
}